File: Services/stockexchange-service/src/main/java/com/cts/project/stockexchangeservice/StockExchangeService.java
package com.cts.project.stockexchangeservice;

import java.util.List;

public interface StockExchangeService {

	public StockExchange addStockExchange(StockExchange stock);
	
	public StockExchange updateStockExchange(StockExchange stock);
	
	public void deleteStockExchange(int id);
	
	public StockExchange getStockExchangeById(int id);
	
	public List<StockExchange> getAllStockExchange();
}

File: StockExchange/src/app/services/stock-exchange.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { StockExchange } from '../models/stock-exchange';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class StockExchangeService {
  httpUrl=environment.url+"stock-exchange-service/stockexchange/";
  constructor(private httpClient:HttpClient) { }
  getAllStockExchange(): Observable<StockExchange[]>{
    return this.httpClient.get<StockExchange[]>(this.httpUrl);
  }
  saveStockExchange(stockexchange: StockExchange): Observable<StockExchange>{
    return this.httpClient.post<StockExchange>(this.httpUrl , stockexchange);
  }
  deleteStockExchange(id: number): Observable<StockExchange> {
    return this.httpClient.delete<StockExchange>(this.httpUrl + id);
  }
  updateStockExchange(stockexchange: StockExchange):Observable<StockExchange>{
    return this.httpClient.put<StockExchange>(this.httpUrl,stockexchange);
    }
    getStockExchangeById(id: number):Observable<StockExchange>{
      return this.httpClient.get<StockExchange>(this.httpUrl+id);
    }
}
